File: NekretnineFrontend/src/app/Models/BaseSearchObject.ts
export class BaseSearchObject{
  public Page : number;
  public PageSize : number;
  public TotalCount : number;
  public TotalPages : number;
  public HasPrevious : boolean;
  public HasNext : boolean;

  constructor(baseSearchObject? : BaseSearchObject,page?: number, pageSize?: number,totalCount?: number, totalPages?: number, HasPrevious ?: boolean, HasNext ?:boolean)
  {
    if(baseSearchObject != null) {
      this.Page = baseSearchObject.Page;
      this.PageSize = baseSearchObject.PageSize;
      this.TotalPages = baseSearchObject.TotalPages;
      this.TotalCount = baseSearchObject.TotalCount;
      this.HasNext = baseSearchObject.HasNext;
      this.HasPrevious = baseSearchObject.HasPrevious;
    }
    else{
      this.Page = page;
      this.PageSize = pageSize;
      this.TotalCount = totalCount;
      this.TotalPages = totalPages
      this.HasNext = HasNext;
      this.HasPrevious = HasPrevious;
    }
  }

}

File: NekretnineAPI/Models/SearchObjects/BaseSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models.SearchObjects
{
    public class BaseSearchObject
    {
        public int? Page { get; set; }
        public int? PageSize { get; set; }
        public int? TotalCount { get; set; }
        public int? TotalPages { get; set; }
        public bool? HasPrevious => Page > 1;
        public bool? HasNext => Page < TotalPages;

    }
}
